signer/src/eth.rs

```diff
@@ -0,0 +1,269 @@
+// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// This file is dual-licensed as Apache-2.0 or GPL-3.0.
+// see LICENSE for license details.
+
+//! An ethereum signer implementation.
+use derive_more::{Display, From};
+use hex::FromHex;
+use keccak_hash::keccak;
+use secp256k1::{Keypair, Message, Secp256k1, SecretKey};
+
+/// An ethereum signer implementation.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EthereumSigner(Keypair);
+
+impl From<crate::ecdsa::Keypair> for EthereumSigner {
+    fn from(kp: crate::ecdsa::Keypair) -> Self {
+        EthereumSigner(kp.0)
+    }
+}
+
+impl EthereumSigner {
+    /// Construct an ethereum signer from a hex-encoded private key.
+    pub fn from_private_key_hex(hex: &str) -> Result<EthereumSigner, Error> {
```

Comment:
Can we unify this API with existing [from_seed, from_uri, from_phrase](https://github.com/paritytech/subxt/blob/master/signer/src/sr25519.rs#L65-#L124) or is it some other reason?

Reply:
i added it. i think its ok not to add test for these since its just a passthrough to the ecdsa module which already have tests.

Reply:
I kind of feel like it would be better if they just did `let keypair: eth::Keypair = ecdsa::Keypair::from_phrase(...).into()` though